ProcessLib/PhaseField/PhaseFieldFEM.h

```diff
@@ -419,20 +338,22 @@ class PhaseFieldLocalAssembler : public PhaseFieldLocalAssemblerInterface
             Ddd.noalias() += N.transpose() / M * N * w;
         }
         // displacement equation, phasefield part
-        local_Jac.template block<displacement_size, phasefield_size>(
-                     displacement_index, phasefield_index)
+        local_Jac
+            .template block<displacement_size, phasefield_size>(
+                displacement_index, phasefield_index)
             .noalias() += Kud;
 
         // phasefield equation, phasefield part.
-        local_Jac.template block<phasefield_size, phasefield_size>(
-                     phasefield_index, phasefield_index)
+        local_Jac
+            .template block<phasefield_size, phasefield_size>(phasefield_index,
+                                                              phasefield_index)
             .noalias() += Kdd + Ddd / dt;
 
         // phasefield equation, displacement part.
-        local_Jac.template block<phasefield_size, displacement_size>(
-                     phasefield_index, displacement_index)
+        local_Jac
+            .template block<phasefield_size, displacement_size>(
+                phasefield_index, displacement_index)
             .noalias() += Kdu;
```

Comment:
@nagelt Above in line 317, in the if-condition (history_prev < strain_energy), the value of `Kdu` is set depending on the last integration point values for the history_prev and strain_energy. Is it correct? Seems to me like the results would depend on the ordering of the integration points.

Reply:
both the history_variable_prev and strain_energy_tensile are integration point data. The if-condition solves Kdu on each integration point.

Reply:
@xingyuanmiao not quite ... the if-condition determines whether or not the term is relevant in each integration point .. that's ok ... but the assigment you make then takes the entire K_ud transpose which has been assembled for all previous integration points (+=) ... so after your if you add more than you need ... and this leads to the randomness Dima describes ...

Reply:
I made the corresponding change in this PR. See commit `[PL] PhaseField: Fix computation of Kdu.`